packages/web5/src/web5.ts

```diff
@@ -84,4 +91,26 @@ export class Web5 {
 
     return { web5, did: connectedDid };
   }
+
+  /**
```

Comment:
this is super cool. 

One suggestion (perhaps future) is to check if the DWN returns 200 OK: 

`
To enhance the code to check that each URL returns a 200 OK status before using it, you can modify the getDwnHosts function as follows:

javascript
Copy code
/**
 * Dynamically selects up to 4 dwn hosts
 */
static async getDwnHosts() {
  const response = await fetch('https://dwn.tbddev.org/.well-known/did.json');

  const didDoc = await response.json();
  const [service] = didUtils.getServices(didDoc, { id: '#dwn', type: 'DecentralizedWebNode' });
  const { nodes } = <DwnServiceEndpoint>service.serviceEndpoint;

  // Allocate up to 4 nodes for a user.
  const numNodesToAllocate = Math.min(Math.floor(nodes.length / 2), 4);
  const dwnUrls = [];

  for (let i = 0; i < numNodesToAllocate; i += 1) {
    let nodeIdx = getRandomInt(0, nodes.length);
    let node = nodes[nodeIdx];
    let nodeResponse = await fetch(node);

    while (nodeResponse.status !== 200) {
      // If the response status is not 200, get another random node
      nodeIdx = getRandomInt(0, nodes.length);
      node = nodes[nodeIdx];
      nodeResponse = await fetch(node);
    }

    dwnUrls.push(node);
  }

  return dwnUrls;
}`

Reply:
this way should a DWN not be available it will "naturally" load balance.

Reply:
Ah yeah great idea @michaelneale. I'll add this in now